socialsharing.php

```diff
@@ -1,277 +1,375 @@
+                    true).'&conf=6&configure='.$this->name.'&tab_module='.$this->tab.'&module_name='.$this->name);
+
+        }
+
+        $helper = new HelperForm();
+        $helper->submit_action = 'submitSocialSharing';
+        $helper->currentIndex = $this->context->link->getAdminLink('AdminModules',
+                false).'&configure='.$this->name.'&tab_module='.$this->tab.'&module_name='.$this->name;
+        $helper->token = Tools::getAdminTokenLite('AdminModules');
+        $helper->tpl_vars = array('fields_value' => $this->getConfigFieldsValues());
+
+        $fields = array();
+        foreach (self::$networks as $network) {
+            $fields[] = array(
+                'type' => 'switch',
+                'label' => $network,
+                'name' => 'PS_SC_'.Tools::strtoupper($network),
+                'values' => array(
+                    array(
+                        'id' => Tools::strtolower($network).'_active_on',
+                        'value' => 1,
+                        'label' => $this->l('Enabled')
+                    ),
+                    array(
+                        'id' => Tools::strtolower($network).'_active_off',
+                        'value' => 0,
+                        'label' => $this->l('Disabled')
+                    )
+                )
+            );
+        }
+
+        return $this->html.$helper->generateForm(array(
+            array(
+                'form' => array(
+                    'legend' => array(
+                        'title' => $this->displayName,
+                        'icon' => 'icon-share'
+                    ),
+                    'input' => $fields,
+                    'submit' => array(
+                        'title' => $this->l('Save')
+                    )
+                )
+            )
+        ));
+    }
+
+    /**
+     * Hook between head tags
+     *
+     * @param array $params Hook parameters
+     * @return string HTML to be displayed
+     */
+    public function hookDisplayHeader($params)
+    {
+        if (!isset($this->context->controller->php_self) || !in_array($this->context->controller->php_self,
+                array('product', 'products-comparison'))
+        ) {
+            return '';
+        }
+
+        $this->context->controller->addCss($this->_path.'css/socialsharing.css');
+        $this->context->controller->addJS($this->_path.'js/socialsharing.js');
+
+        if ($this->context->controller->php_self == 'product') {
+            $product = $this->context->controller->getProduct();
+
+            if (!Validate::isLoadedObject($product)) {
+                return '';
+            }
+            if (!$this->isCached('socialsharing_header.tpl',
+                $this->getCacheId('socialsharing_he ader|'.
+                    (isset($product->id) && $product->id ? (int)$product->id : '')))
+            ) {
+                if (!defined('_PS_PRICE_COMPUTE_PRECISION_')) {
```

Comment:
... but if shops are updated to 1.6.1.5, this constant is defined so ... we dont need to do a check.

The only - and very good - solution is to update version of PrestaShop in concerned websites, right ?